greentest/test__issues461_471.py

```diff
@@ -0,0 +1,42 @@
+'''Test for GitHub issues 461 and 471.
+
+When moving to Python 3, handling of KeyboardInterrupt exceptions caused
+by a Ctrl-C raise an exception while printing the traceback for a
+greenlet preventing the process from exiting. This test tests for proper
+handling of KeyboardInterrupt.
+'''
+
+import sys
+
+if sys.argv[1:] == ['subprocess']:
+    import gevent
+
+    def task():
+        sys.stdout.write('ready\n')
+        sys.stdout.flush()
+        gevent.sleep(30)
+    try:
+        gevent.spawn(task).get()
+    except KeyboardInterrupt:
+        pass
+else:
+    import signal
+    from subprocess import Popen, PIPE
+    import time
+
+    if sys.platform.startswith('win'):
+        from subprocess import CREATE_NEW_PROCESS_GROUP
+        kwargs = {'creationflags': CREATE_NEW_PROCESS_GROUP}
+    else:
+        kwargs = {}
+    p = Popen([sys.executable, __file__, 'subprocess'], stdout=PIPE, **kwargs)
+    p.stdout.readline()
+    p.send_signal(signal.SIGINT)
+    for i in range(30):
+        if p.poll() is not None:
+            break
+        time.sleep(0.1)
+    else:
+        p.terminate()
```

Comment:
On POSIX-compliant platforms, `terminate()` just sends a signal. We should still `wait()` after after that to make sure the child has really exited, It's best practice, like `join()` is in case of greenlets, threads, and `multiprocessing.Process` objects.